### ISSUE TITLE
POE Switching

### ISSUE THREAD

**Issue by Elespecia** [zachcheatham/ha-omada#27]
I might be out on thin ice requesting features but one major feature would be to have the POE-ports as binary switches.
https://community.tp-link.com/en/business/forum/topic/104614
Gotta learn coding so I can contribute for real.

**Comment by erwinvana** [comment]
I second this request.

**Comment by baylanger** [comment]
@erwinvana is your switch for sale?  I’m waiting for mine, back order 😂 

I haven’t checked the code of this project yet, I’ve figured I’d do that when I get my switch.  It might be very simple to add.  Not sure how busy @Elespecia is these days but he offered to help.  If you have free time and know what you’re doing, he can likely work with you.

**Comment by baylanger** [comment]
@Elespecia sorry I tag you… instead of @zachcheatham

**Comment by ElStupid** [comment]
Really would love that too. To manage energy consumption while I'm away or asleep, I would like to be able to switch off AP's.
Keep up the good work @zachcheatham !

**Comment by ZdenekM** [comment]
Also might be useful to reboot or recover unreliable devices like Chinese cameras... So far, I have to call poe recover from Omada Controller manually but it would be great if HA can do it for me :)

**Comment by zachcheatham** [comment]
I would love to add this feature, but I do not have an Omada switch in my network to test with. If anyone with a switch wants to take on this feature, it would be much appreciated.

**Comment by star3141** [comment]
Hi!

I would love that feature too. Alternatively, I am trying to figure out how to do that by means of SNMP but, unfortunately, without success at the moment.

**Comment by tensiondriven** [comment]
Forgive me for asking a question unrelated to implementation of the specific feature; I'm curious, what is the use case for wanting to turn off power to a PoE port?  I've never run into this use case, nor can I think of one... surely im missing something

**Comment by baylanger** [comment]
1. Save Energy
2. Potentially increase life expectancy of PoE port & switch's power ACDC adapter and the PoE devices 

. security cameras, indoor or outdoor are not always needed 7/24.
. outdoor / far / office wifi spot not needed at night
. VoIP phone, not at night either - only after I had my coffe :p
. any devices powered on via PoE over ~5-24/VDC (small hubs, IoT, RasberryPi, etc)

With price of electricity and cost of switches/devices, seems this would be a nice green option to have.

That said. I have 2 Netgear switch, the last one I bought was a Pro. There's no API or console access. I started to looking around at options that can allow me to do it. My current fallback is a Cisco 300 Series and can do it over ssh.

My dream is to have home-assistant call an API to get the status of the PoE for each port and be able as well to control them.

**Comment by baylanger** [comment]
Can also be useful if a device becomes unresponsive (ping) .. or unstable video stream from camera/bad image...

My router (mikrotik) facing my ISP is powered up as well over PoE. Currently anything on PoE that freezes, it's faster to go downstairs then to ssh.

**Comment by wuppiwuppi** [comment]
Yes - if a PoE powered RaspberryPi or other device becomes unresponsive -> being able to power cycle it would be nice. 

For me it is a power-saving issue.

**Comment by TommySharpNZ** [comment]
This is pretty much the one feature of the "official" Omada integration at the moment. 
Works well so I have both integrations loaded :-)

**Comment by gmoscalu** [comment]
Hello,

Capturing Omada GUI API call on a TL-SG2428P POE switch:

https://<OMADA_HOST>:<OMADA_PORT>/<CID>/api/v2/sites/<SITE_ID>/cmd/switches/<DEV_MAC>/ports/<SW_PORT_ID>/restart

Captured this while doing a "PoE Recovery"  on an AP that went away (disconnected).

Hope this helps,
Gabriel
